router.ts

```diff
@@ -0,0 +1,165 @@
+import { isEmpty, isFunction, Status, STATUS_TEXT } from "./deps.ts";
+
+/** HTTP request method. */
+export type Method =
+  /** RFC 9110, 9.3.1 */
+  | "GET"
+  /** RFC 9110, 9.3.2 */
+  | "HEAD"
+  /** RFC 9110, 9.3.3 */
+  | "POST"
+  /** RFC 9110, 9.3.4 */
+  | "PUT"
+  /** RFC 9110, 9.3.5 */
+  | "DELETE"
+  /** RFC 9110, 9.3.6 */
+  | "CONNECT"
+  /** RFC 9110, 9.3.7 */
+  | "OPTIONS"
+  /** RFC 9110, 9.3.8 */
+  | "TRACE"
+  /** RFC 5789 */
+  | "PATCH"
+  // deno-lint-ignore ban-types
+  | ({} & string);
+
+/** HTTP request router API. */
+export interface Router {
+  (req: Request): Response | Promise<Response>;
+}
+
+/**
+ * A handler for HTTP route requests. Consumes a request and connection information
+ * and returns a response.
+ *
+ * If a handler throws, the server calling the handler will assume the impact
+ * of the error is isolated to the individual request. It will catch the error
+ * and close the underlying connection.
+ */
+export type RouteHandler = (
+  req: Request,
+  params: { readonly [k in string]?: string },
+) => Promise<Response> | Response;
+
+/** HTTP router routes. */
+export interface Routes {
+  readonly [k: string]: RouteHandler | MethodRouteHandlers;
+}
+
+/** Create router options. */
+export interface Options {
+  /** If a `GET` handler is defined, it will be used to generate a response to the `HEAD` request.
+   *
+   * This feature is based on RFC 9110, 9.1
+   * > All general-purpose servers MUST support the methods GET and HEAD.
+   * @default true
+   */
+  withHead: boolean;
+}
+
+/** Map for HTTP method and {@link RouteHandler} */
+export type MethodRouteHandlers = { [k in Method]?: RouteHandler };
+
+function methods(
+  methodRouteHandlers: Readonly<MethodRouteHandlers>,
+): RouteHandler {
+  return (req, params) => {
+    const routeHandler = methodRouteHandlers[req.method];
+    if (routeHandler) {
+      return routeHandler(req, params);
+    }
+
+    const allows = Object.keys(methodRouteHandlers);
+
+    return new Response(null, {
+      status: Status.MethodNotAllowed,
+      statusText: STATUS_TEXT[Status.MethodNotAllowed],
+      headers: {
+        allow: allows.join(","),
+      },
+    });
+  };
+}
+
+/** Create HTTP request router.
+ *
+ * ```ts
+ * import { createRouter } from "https://deno.land/x/http_router@$VERSION/mod.ts";
+ * import { serve } from "https://deno.land/std@$VERSION/http/mod.ts";
+ * const router = createRouter({
+ *   "/api/students/:name": {
+ *     GET: (req, params) => {
+ *       const greeting = `Hello! ${params.name!}`;
+ *       return new Response(greeting);
+ *     },
+ *   },
+ *   "/api/status": () => new Response("OK"), // Any HTTP request method
+ * });
+ * await serve(router);
+ * ```
+ * @throws TypeError
+ * - The given route path is invalid url path.
+ */
+export function createRouter(
+  routes: Routes,
+  { withHead = true }: Partial<Options> = {},
```

Comment:
Codacy has a fix for the issue: [Delete `,`](https://app.codacy.com/gh/httpland/http-router/pullRequest?prid=9963312)
```suggestion
  { withHead = true }: Partial<Options> = {}
```